src/sentry/api/endpoints/project_rules.py

```diff
@@ -61,83 +62,94 @@ def post(self, request: Request, project) -> Response:
         )
-
-        if serializer.is_valid():
-            data = serializer.validated_data
-            # combine filters and conditions into one conditions criteria for the rule object
-            conditions = data.get("conditions", [])
-            if "filters" in data:
-                conditions.extend(data["filters"])
-
-            kwargs = {
-                "name": data["name"],
-                "environment": data.get("environment"),
-                "project": project,
-                "action_match": data["actionMatch"],
-                "filter_match": data.get("filterMatch"),
-                "conditions": conditions,
-                "actions": data.get("actions", []),
-                "frequency": data.get("frequency"),
-                "user_id": request.user.id,
-            }
-            owner = data.get("owner")
-            if owner:
-                try:
-                    kwargs["owner"] = owner.resolve_to_actor().id
-                except (User.DoesNotExist, Team.DoesNotExist):
-                    return Response(
-                        "Could not resolve owner",
-                        status=status.HTTP_400_BAD_REQUEST,
-                    )
-
-            if data.get("pending_save"):
-                client = RedisRuleStatus()
-                uuid_context = {"uuid": client.uuid}
-                kwargs.update(uuid_context)
-                find_channel_id_for_rule.apply_async(kwargs=kwargs)
-                return Response(uuid_context, status=202)
-
-            created_alert_rule_ui_component = trigger_sentry_app_action_creators_for_issues(
-                kwargs.get("actions")
-            )
-            rule = project_rules.Creator.run(request=request, **kwargs)
-            RuleActivity.objects.create(
-                rule=rule, user_id=request.user.id, type=RuleActivityType.CREATED.value
-            )
-            duplicate_rule = request.query_params.get("duplicateRule")
-            wizard_v3 = request.query_params.get("wizardV3")
-
-            self.create_audit_entry(
-                request=request,
-                organization=project.organization,
-                target_object=rule.id,
-                event=audit_log.get_event_id("RULE_ADD"),
-                data=rule.get_audit_log_data(),
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+        # combine filters and conditions into one conditions criteria for the rule object
+        conditions = data.get("conditions", [])
+        if "filters" in data:
+            conditions.extend(data["filters"])
+
+        rules = Rule.objects.filter(project=project, status=RuleStatus.ACTIVE)
+        slow_rules = 0
+        for rule in rules:
+            for condition in rule.data["conditions"]:
+                for slow_conditions in SLOW_CONDITION_MATCHES:
+                    if slow_conditions in condition["id"]:
```

Comment:
Might be worth breaking this out into a `is_condition_slow` function that we can share with the rule processor